specification/vmware/Microsoft.AVS/routes.tsp

```diff
@@ -0,0 +1,1362 @@
+      ...PrivateCloudNameParameter;
+    },
+    WorkloadNetworkList
+  >;
+
+  @doc("List of segments in a private cloud workload network.")
+  @route("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AVS/privateClouds/{privateCloudName}/workloadNetworks/default/segments")
+  @get
+  listSegments is Azure.Core.Foundations.Operation<
+    {
+      ...SubscriptionIdParameter;
+      ...ResourceGroupParameter;
+      ...PrivateCloudNameParameter;
+    },
+    WorkloadNetworkSegmentsList
+  >;
+
+  @doc("Get a segment by id in a private cloud workload network.")
+  @route("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AVS/privateClouds/{privateCloudName}/workloadNetworks/default/segments/{segmentId}")
+  @get
+  getSegment is Azure.Core.Foundations.Operation<
+    {
+      ...SubscriptionIdParameter;
+      ...ResourceGroupParameter;
+      ...PrivateCloudNameParameter;
+      ...SegmentIdParameter;
+    },
+    WorkloadNetworkSegment
+  >;
+
+  @doc("Create a segment by id in a private cloud workload network.")
+  @route("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AVS/privateClouds/{privateCloudName}/workloadNetworks/default/segments/{segmentId}")
+  @put
+  createSegments is Azure.Core.Foundations.Operation<
+    {
+      ...SubscriptionIdParameter;
+      ...ResourceGroupParameter;
+      ...PrivateCloudNameParameter;
+      ...SegmentIdParameter;
+
+      @doc("NSX Segment")
+      @body
+      workloadNetworkSegment: WorkloadNetworkSegment;
+    },
+    WorkloadNetworkSegment
+  >;
+
+  @doc("Create or update a segment by id in a private cloud workload network.")
+  @route("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AVS/privateClouds/{privateCloudName}/workloadNetworks/default/segments/{segmentId}")
+  @patch
+  updateSegments is Azure.Core.Foundations.Operation<
+    {
+      ...SubscriptionIdParameter;
+      ...ResourceGroupParameter;
+      ...PrivateCloudNameParameter;
+      ...SegmentIdParameter;
+
+      @doc("NSX Segment")
+      @body
+      workloadNetworkSegment: WorkloadNetworkSegment;
+    },
+    WorkloadNetworkSegment | void
+  >;
+
+  @doc("Delete a segment by id in a private cloud workload network.")
+  @route("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AVS/privateClouds/{privateCloudName}/workloadNetworks/default/segments/{segmentId}")
+  @delete
+  deleteSegment is Azure.Core.Foundations.Operation<
+    {
+      ...SubscriptionIdParameter;
+      ...ResourceGroupParameter;
+      ...PrivateCloudNameParameter;
+      ...SegmentIdParameter;
+    },
+    void
+  >;
+
+  @doc("List dhcp in a private cloud workload network.")
+  @route("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AVS/privateClouds/{privateCloudName}/workloadNetworks/default/dhcpConfigurations")
```

Comment:
workloadNetworks should be modeld as a singleton resource: https://azure.github.io/typespec-azure/docs/howtos/ARM/resource-type#singleton-resource

Reply:
And dhcpConfiguration should be a child of this resource